File: Test/AdventureWorksFunctionalModel/Mapping/ShiftMap.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Data.Entity.ModelConfiguration;

using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;

namespace AW.Mapping {
    public class ShiftMap : EntityTypeConfiguration<Shift> {
        public ShiftMap() {
            // Primary Key
            HasKey(t => t.ShiftID);

            // Properties
            Property(t => t.Name)
                .IsRequired()
                .HasMaxLength(50);

            // Table & Column Mappings
            ToTable("Shift", "HumanResources");
            Property(t => t.ShiftID).HasColumnName("ShiftID").HasDatabaseGeneratedOption(DatabaseGeneratedOption.Identity);
            Property(t => t.Name).HasColumnName("Name");
            Property(t => t.StartTime).HasColumnName("StartTime");
            Property(t => t.EndTime).HasColumnName("EndTime");
            Property(t => t.ModifiedDate).HasColumnName("ModifiedDate"); //.IsConcurrencyToken();
        }
    }

    public static partial class Mapper {
        public static void Map(this EntityTypeBuilder<Shift> builder) {
            builder.HasKey(t => t.ShiftID);

            // Properties
            builder.Property(t => t.Name)
                   .IsRequired()
                   .HasMaxLength(50);

            // Table & Column Mappings
            builder.ToTable("Shift", "HumanResources");
            builder.Property(t => t.ShiftID).HasColumnName("ShiftID").ValueGeneratedOnAdd();
            builder.Property(t => t.Name).HasColumnName("Name");
            builder.Property(t => t.StartTime).HasColumnName("StartTime");
            builder.Property(t => t.EndTime).HasColumnName("EndTime");
            builder.Property(t => t.ModifiedDate).HasColumnName("ModifiedDate"); //.IsConcurrencyToken();
        }
    }
}

File: Test/AdventureWorksLegacyModelVB/Mapping/ShiftMap.vb
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.Data.Entity.ModelConfiguration

Imports Microsoft.EntityFrameworkCore
Imports Microsoft.EntityFrameworkCore.Metadata.Builders

Namespace AW.Mapping
	Public Class ShiftMap
		Inherits EntityTypeConfiguration(Of Shift)

		Public Sub New()
			' Primary Key
			HasKey(Function(t) t.ShiftID)

			' Properties
			[Property](Function(t) t.Name).IsRequired().HasMaxLength(50)

			' Table & Column Mappings
			ToTable("Shift", "HumanResources")
			[Property](Function(t) t.ShiftID).HasColumnName("ShiftID").HasDatabaseGeneratedOption(DatabaseGeneratedOption.Identity)
			[Property](Function(t) t.Name).HasColumnName("Name")
			[Property](Function(t) t.StartTime).HasColumnName("StartTime")
			[Property](Function(t) t.EndTime).HasColumnName("EndTime")
			[Property](Function(t) t.ModifiedDate).HasColumnName("ModifiedDate") '.IsConcurrencyToken();
		End Sub
	End Class

	Public Module Mapper
		<System.Runtime.CompilerServices.Extension> _
		Public Sub Map(ByVal builder As EntityTypeBuilder(Of Shift))
			builder.HasKey(Function(t) t.ShiftID)

			' Properties
			builder.Property(Function(t) t.Name).IsRequired().HasMaxLength(50)

			' Table & Column Mappings
			builder.ToTable("Shift", "HumanResources")
			builder.Property(Function(t) t.ShiftID).HasColumnName("ShiftID").ValueGeneratedOnAdd()
			builder.Property(Function(t) t.Name).HasColumnName("Name")
			builder.Property(Function(t) t.StartTime).HasColumnName("StartTime")
			builder.Property(Function(t) t.EndTime).HasColumnName("EndTime")
			builder.Property(Function(t) t.ModifiedDate).HasColumnName("ModifiedDate") '.IsConcurrencyToken();
		End Sub
	End Module
End Namespace